pkg/apis/alerting/v1/extensions.go

```diff
@@ -355,3 +417,35 @@ func (r *ListRoutingRelationshipsResponse) GetInvolvedConditions(endpointId stri
 	}
 	return involvedConditions
 }
+func (l *ListAlertConditionRequest) FilterFunc() func(*AlertCondition, int) bool {
+	return func(item *AlertCondition, _ int) bool {
+		if len(l.Clusters) != 0 {
+			if !slices.Contains(l.Clusters, item.GetClusterId().Id) {
+				return false
+			}
+		}
+		if len(l.Labels) != 0 {
+			if len(lo.Intersect(l.Labels, item.Labels)) != len(l.Labels) {
```

Comment:
This is going to allocate memory for a new map, I'd do this check by hand instead